include/hobbes/eval/cc.H

```diff
@@ -287,20 +276,20 @@ private:
   SearchCache searchCache;
 
   // optimization options
-  bool runModInlinePass;
-  bool genInterpretedMatch;
-  bool checkMatchReachability;
-  bool lowerPrimMatchTables;
-  bool columnwiseMatches;
-  size_t maxExprDFASize={1000};
+  bool runModInlinePass       = true;
```

Comment:
how are these changes related to the eliminators memory leak issue?

Reply:
No, this is about modernization. Again, I really should have split this commit

`cc` already has enough member variables, and I just added another one in #408. init list in constructor before c++11 is easy to miss a member or get the order wrong.

That's why I think direct member initialization is a better choice in most cases, won't left member uninitialized and don't need to worry about initialization order

Reply:
Agreed, let's split.

Btw, aren't there any compiler flags to help catch members initialization order issues? Cam we have the compiler catch these instead?

Reply:
I checked the git log of CMakeLists.txt, interestingly, `-Wreorder` was the first warning flag added explicitly. So, yes, as you pointed out, this kind of error will be caught.

However, still nothing prevents me from making this kind of mistake

```cpp
struct S {
    S() : b(0) {}
private:
    int a;
    int b;
};
```
[gotbolt](https://godbolt.org/z/qTaedqTGv)